Changed CAD Export to not export configuration elements without visualization category

PhilMFischer opened the pull request:
Based on #206 therefore #208 should be closed and reviewed before.

Now using the new BeanTraverser of #206 to create the Json Object. Also adjusted the test cases.

Closes #204

Commit:
Added isRoot method to bean SEI - (Task #204)

- Added test method to see if SEI is a root element to the bean API
- Added corresponding test case as well.
- Cleanup the code in a few spots.

---
Task #204: CAD Exports Structural Elemets without Visualization Category

Commit:
Created the initial API for Bean Traverser - (Task #206)

[skip ci]
---
Task #206: Create a generic Bean Traverser

Commit:
Bean Traverser API simplification - (Task #206)

[skip ci]
---
Task #206: Create a generic Bean Traverser

Commit:
Added Bean Traverser tests - (Task #206)

[skip ci]
---
Task #206: Create a generic Bean Traverser

Commit:
Implemented BeanStructuralTreeTraverser - (Task #206)

Moved its tests to test concept

---
Task #206: Create a generic Bean Traverser

Commit:
Updated mailmap file - (Task #206)

---
Task #206: Create a generic Bean Traverser

Commit:
Merge remote-tracking branch 'origin/feature/task_206-Create_a_generic_Bean_Traverser' into feature/task_204-CAD_Exports_Structura

Commit:
Fixed issue with placing children attribute - (Task #206)

- Placed children into themself and not in the parent as expected.
[skip ci]

---
Task #206: Create a generic Bean Traverser

Commit:
Fixed CatiaExporter tests - (Task #206)

---
Task #206: Create a generic Bean Traverser

andriikovalov-dlr commented on de.dlr.sc.virsat.model.edit/src/de/dlr/sc/virsat/model/concept/types/structural/IBeanStructuralElementInstance.java:
```diff
@@ -237,4 +237,10 @@
 	<BEAN_TYPE extends IBeanCategoryAssignment> boolean hasCategory(Class<BEAN_TYPE> catBeanClazz);
+	
+	/**
+	 * Checks if the current Bean SEI is a root element.
```
But this is actually not what's being checked - you check if this SEI is of a SE that can be a root element. Let's say in CEF for a System under SystemOfSystems it will return true although it's not a root element

andriikovalov-dlr commented on de.dlr.sc.virsat.model.extension.mechanical/src/de/dlr/sc/virsat/model/extension/mechanical/catia/CatiaExporter.java:
```diff
@@ -117,41 +120,61 @@ public JsonObject transformElement(IBeanStructuralElementInstance element) {
+			public boolean isMatching(IBeanStructuralElementInstance treeNode) {
+				visCa = treeNode.getFirst(Visualisation.class);
+				hasVisCa = visCa != null;
+				isRoot = treeNode.isRoot();
```
Should we maybe check `treeNode == productBean` for root? This would correctly handle exporting a subtree.

SaMuellerDLR commented on de.dlr.sc.virsat.model.edit/src/de/dlr/sc/virsat/model/concept/types/structural/tree/BeanStructuralTreeTraverser.java:
```diff
@@ -0,0 +1,51 @@
+			matcher.foundMatch(node, parent);
+			nextParent = node;
+		}
+		for (IBeanStructuralElementInstance child: node.getChildren(IBeanStructuralElementInstance.class)) {
```
`for (IBeanStructuralElementInstance child: node.getChildren(IBeanStructuralElementInstance.class)) {`
whitespacing should be
`for (IBeanStructuralElementInstance child : node.getChildren(IBeanStructuralElementInstance.class)) {`

SaMuellerDLR commented on de.dlr.sc.virsat.model.extension.tests.test/src/de/dlr/sc/virsat/model/concept/types/structural/tree/BeanStructuralTreeTraverserTest.java:
```diff
@@ -0,0 +1,101 @@
+		BeanStructuralTreeTraverser traverser = new BeanStructuralTreeTraverser();
+		traverser.traverse(n, matcher);
+		
+		final int THREE = 3;
```
I would prefer the a name like `EXPECTED_MATCHES` instead of THREE.
Making a constant for a magic number with the name of the magic number just sidesteps checkstyle but doesnt tell what its supposed to mean.

SaMuellerDLR replied:
You fixed the issue in testAllMatches which I actually overlooked, but didnt fix it in testSomeMatch :D

andriikovalov-dlr committed:
Improved constant name - (Task #204)

Minor whitespace fix

---
Task #204: CAD Exports Structural Elemets without Visualization Category

andriikovalov-dlr committed:
Renamed isRoot into canBeRoot - (Task #204)

Changed CatiaExporter to correctly handle root element when exporting
subtree

---
Task #204: CAD Exports Structural Elemets without Visualization Category

andriikovalov-dlr committed:
Renamed another THREE constant in test - (Task #204)

---
Task #204: CAD Exports Structural Elemets without Visualization Category

SaMuellerDLR reviewed (dismissed):
Approved!

andriikovalov-dlr committed:
Trying to get codecov notification - (Task #204)

---
Task #204: CAD Exports Structural Elemets without Visualization Category

SaMuellerDLR reviewed (approved):
Approved!
